### PR TITLE
Начинаем програмировать

### PR DESCRIPTION
**Pull request by juzeppe13**
---
:mortar_board: [Начинаем програмировать](https://up.htmlacademy.ru/ecmascript/8/user/1001479/tasks/2)

:boom: https://htmlacademy-ecmascript.github.io/1001479-taskmanager-8/2/

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+    acc += createFilter(item, count, index === 0);
+
+    return acc;
+  }, ``);
+};
+
+addFilters();
+
+const createCard = (color, type, text) => `
```
Comment (sasZilla) [inline src/main.js]: use object {color, type, text} in arguments instead
it simplify adding new params in future

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        />
+        <label for="filter__${convertLabelToId(label)}" class="filter__label">
+          ${label} <span class="filter__${convertLabelToId(label)}-count">${count}</span></label
+        >`;
+};
+
+const addFilters = () => {
+  const filtersSection = document.querySelector(`.main__filter`);
+
+  filtersSection.innerHTML = filters.reduce((acc, item, index) => {
```
Comment (sasZilla) [inline src/main.js]: I like your style using reducers

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+const getRandomText = () => [`some text here`, `another text there`, `text is everywhere`][randomInteger(0, 2)];
+
+const addCards = (count) => {
+  const cardsSection = document.querySelector(`.board__tasks`);
+  let innerHtml = ``;
+
+  for (let i = 0; i < count; i++) {
+    innerHtml += createCard(getRandomCardColor(), getRandomCardModifier(), getRandomText());
+  }
+  cardsSection.innerHTML = innerHtml;
```
Comment (sasZilla) [inline src/main.js]: looks you know why concatenation before inserting in Dom important, that's cool

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,351 @@
+'use strict';
+
+const filters = [`All`, `Overdue`, `Today`, `Favorites`, `Repeating`, `Tags`, `Archive`];
```
Comment (sasZilla) [inline src/main.js]: could you create kind of global settings object to put all common information there
like const settings = { filters, ... }

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+
+  for (let i = 0; i < count; i++) {
+    innerHtml += createCard(getRandomCardColor(), getRandomCardModifier(), getRandomText());
+  }
+  cardsSection.innerHTML = innerHtml;
+};
+
+addCards(7);
+
+function randomInteger(min, max) {
```
Comment (sasZilla) [inline src/main.js]: this looks like external/common helper function, you could create helper module as different file or section in you current file, but don't forget about comments to explain it

File: src/main.js [hunk 6]
```diff
@@ -0,0 +1,351 @@
+'use strict';
+
+const filters = [`All`, `Overdue`, `Today`, `Favorites`, `Repeating`, `Tags`, `Archive`];
+
+const createFilter = (label, count, isChecked = false) => {
+  const addChecked = (val) => val ? `checked` : ``;
+  const convertLabelToId = (val) => val.toLowerCase().replace(/ /g, `-`);
+
+  return `<input
```
[the remark this answers is not in the document]
Comment (juzeppe13) [reply src/main.js]: current project markup with its css rules based on this structure. Also it has an id and for parameter on a label. Not sure is it on my responsability :)

**Pull request merged**